src/main.ts

```diff
@@ -1,27 +1,31 @@
 'use strict';
-const url = require('url');
-const micro = require('micro');
+import url from 'url';
+import { Server, IncomingMessage, ServerResponse } from 'http'
+import serve, { send } from 'micro';
 const metrics = require('./lib/metrics');
-const { logger } = require('./lib/logger');
-const { Exporter } = require('./lib/kafka_storage');
+import { logger } from './lib/logger';
+import { Exporter } from './lib/kafka_storage';
 const EXPORTER_NAME = process.env.EXPORTER_NAME || 'san-chain-exporter';
-const { BLOCKCHAIN, EXPORT_TIMEOUT_MLS } = require('./lib/constants');
-const worker = require(`./blockchains/${BLOCKCHAIN}/${BLOCKCHAIN}_worker`);
-const constants = require(`./blockchains/${BLOCKCHAIN}/lib/constants`);
-const constantsBase = require('./lib/constants');
+import { EXPORT_TIMEOUT_MLS } from './lib/constants';
+import { constructWorker } from './blockchains/construct_worker'
+import constantsBase from './lib/constants';
+import { ExporterPosition } from './types'
+import { BaseWorker } from './lib/worker_base';
+
+export class Main {
```

Comment:
Move a lot of code which used to be in index.js into a Main class. This helps testing as the test do not need to include the index which triggers `main()`.